src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -995,7 +995,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         if (sym != null && sym.isDeprecated)
           context.deprecationWarning(tree.pos, sym)
         // Keep the original tree in an annotation to avoid losing tree information for plugins
-        treeCopy.Literal(tree, value).updateAttachment(OriginalTreeAttachment(original))
+        if (tree.hasAttachment[OriginalTreeAttachment]) treeCopy.Literal(tree, value)
+        else treeCopy.Literal(tree, value).updateAttachment(OriginalTreeAttachment(tree))
```

Comment:
shouldn't this be be adding `OriginalTreeAttachment(original)`?

Reply:
`tree == original` in this case, but `original` is a parameter off the enclosing gigantic method while `tree` is the parameter of `adaptType`. I opted for `tree` because it's closer to the use site, and because the original tree from which we copy is `tree` and not `original`.